viite-UI/src/view/LinkPropertyLayer.js

```diff
@@ -375,21 +364,28 @@
       });
     };
 
-    /**
-     * Simple method that will remove various open layers 3 features from a selection.
-     * @param features
-     */
-    var removeFeaturesFromSelection = function (features) {
-      var olUids = _.map(selectSingleClick.getFeatures().getArray(), function (feature) {
-        return feature.ol_uid;
-      });
-      _.each(features, function (feature) {
-        if (_.includes(olUids, feature.ol_uid)) {
-          selectSingleClick.getFeatures().remove(feature);
-          olUids.push(feature.ol_uid);
-        }
-      });
+    var getSelectedId = function (selected) {
+      if (!_.isUndefined(selected.id) && selected.id > 0) {
+        return selected.id;
+      } else {
+        return selected.linkId;
+      }
+    };
+
 
+    var addToPreviousSelection = function (ctrlPressed, selection) {
+       if (ctrlPressed && !_.isUndefined(selectedLinkProperty.get()) && !_.isUndefined(selection)) {
+         var selectedLinkIds = _.map(selectedLinkProperty.get(), function (selected) {
+           return getSelectedId(selected);
+         });
+         if (_.includes(selectedLinkIds, getSelectedId(selection))) {
+           selectedLinkIds = _.without(selectedLinkIds, getSelectedId(selection));
+         } else {
+           selectedLinkIds = selectedLinkIds.concat(getSelectedId(selection));
+         }
+         var features = getAllFeatures(true, true, true, true, true, true, true, true, true);
```

Comment:
(Vähän of topic, mutta sanonpa silti: Huutelee refaktorointia, kun 1) iso kasa parametreja 2) funkkari väittää että getAll---, ja sitten kyselee kuitenkin kasan booleaneja. Kohta 1) pätee ja on peräisin sisarfunktiosta getVisibleFeatures.)